### PR TITLE
Solutions Added

### PR DESCRIPTION
**Pull request by kavanbrandon**
Hi Al and Nicolas, 

I've add my solutions to the Swift Challenge in the attached Pull Request. Please let me know if you have any questions. 

Thanks,
Kavan

### CODE REVIEW

File: MakeSpace Swift Challenge April 2015.playground/section-1.swift [hunk 1]
```diff
@@ -4,35 +4,144 @@ import UIKit
 
 //The Swift standard library implements the common CS function "map".  Implement map using generics.  In comments tell us why using generics is "better" than using explicit types
 
-func makespaceMap<A, B>(xs: [A], f: A -> B) -> [B] {
+//Solution 1
+func makespaceMap<A,B>(xs: [A], f: A -> B) -> [B] {
+    
+    return map(xs, f)
```
Comment (altyus) [inline MakeSpace Swift Challenge April 2015.playground/section-1.swift]: We're looking for you to reimplement map.  Try this again without using the Swift map function.

File: MakeSpace Swift Challenge April 2015.playground/section-1.swift [hunk 2]
```diff
@@ -4,35 +4,144 @@ import UIKit
...
 
 //The Swift standard library also has an implementation for the "reduce" function.  Reimplement reduce using iteration.
 
+//Solution 1
 func reduceIteratively <A, R>(xs: [A], initialValue: R, combine: (R, A) -> R) -> R {
+    
+    var result = initialValue
+    
+    for i in xs {
+        result = reduce(xs, initialValue, combine)
```
Comment (altyus) [inline MakeSpace Swift Challenge April 2015.playground/section-1.swift]: Try this again without using swift's reduce function

File: MakeSpace Swift Challenge April 2015.playground/section-1.swift [hunk 3]
```diff
@@ -4,35 +4,144 @@ import UIKit
...
+
+//-------------------------
+
+// MARK: - Recursive Reduce
 
 //Implement reduce recursively.  You are not allowed to use any loops or iterators in this implementation.
 
 func reduceRecursively <A, R>(xs: [A], initialValue: R, combine:(R, A) -> R) -> R {
+    
+    return reduce(xs, initialValue, combine)
```
Comment (altyus) [inline MakeSpace Swift Challenge April 2015.playground/section-1.swift]: Try this function without using the Swift reduce function.  Also make sure the call is recursive.

File: MakeSpace Swift Challenge April 2015.playground/section-1.swift [hunk 4]
```diff
@@ -4,35 +4,144 @@ import UIKit
...
+////Implement the classic fizzbuzz problem.  Your function takes in a start integer and an end integer.  Your function should return an array with integers replaced with Fizz if they are divisible by 3, Buzz if they are divisible by 5, FizzBuzz if they are divisible by 3 and 5. Otherwise the integer should be returned.
+//
 
+//Solution
 func fizzBuzz(start: Int, end: Int) -> [Any]{
+    
+    var result: [Any] = []
+    
+    for i in start...end {
+        if i % 3 == 0 && i % 5 == 0 {
```
Comment (altyus) [inline MakeSpace Swift Challenge April 2015.playground/section-1.swift]: How could you simplify this conditional?

File: MakeSpace Swift Challenge April 2015.playground/section-1.swift (hunk 1 above)
Comment (kavanbrandon) [inline MakeSpace Swift Challenge April 2015.playground/section-1.swift]: Hi Al,

I corrected this solution to implement map without using map function.

File: MakeSpace Swift Challenge April 2015.playground/section-1.swift (hunk 2 above)
Comment (kavanbrandon) [inline MakeSpace Swift Challenge April 2015.playground/section-1.swift]: I added two versions of the solution. I found this blog post to be very helpful ( http://natashatherobot.com/functional-swift-tail-recursion/)

File: MakeSpace Swift Challenge April 2015.playground/section-1.swift (hunk 3 above)
Comment (kavanbrandon) [inline MakeSpace Swift Challenge April 2015.playground/section-1.swift]: I added two versions of the solution. I found this blog post to be very helpful ( http://natashatherobot.com/functional-swift-tail-recursion/)

File: MakeSpace Swift Challenge April 2015.playground/section-1.swift (hunk 4 above)
Comment (kavanbrandon) [inline MakeSpace Swift Challenge April 2015.playground/section-1.swift]: I changed conditional to  "if i % 15 == 0"

File: MakeSpace Swift Challenge April 2015.playground/section-1.swift (hunk 2 above)
Comment (kavanbrandon) [inline MakeSpace Swift Challenge April 2015.playground/section-1.swift]: Hi Al, 
In Solution 2 for this question, I didn't use the reduce function. Did you want me to take another attempt and try to solve a different way? Thanks!

**Pull request closed without merging**